src/Microsoft.OpenApi/Services/OpenApiUrlTreeNode.cs

```diff
@@ -235,5 +236,152 @@ public void AddAdditionalData(Dictionary<string, List<string>> additionalData)
+            { "OTHER", new MermaidNodeStyle("White", MermaidNodeShape.SquareCornerRectangle) },
+        };
+        
+        private static void ProcessNode(OpenApiUrlTreeNode node, StreamWriter writer)
+        {
+            var path = string.IsNullOrEmpty(node.Path) ? "/" : SanitizeMermaidNode(node.Path);
+            var methods = GetMethods(node);
+            var (startChar, endChar) = GetShapeDelimiters(methods);
+            foreach (var child in node.Children)
+            {
+                var childMethods = GetMethods(child.Value);
+                var (childStartChar, childEndChar) = GetShapeDelimiters(childMethods);
+                writer.WriteLine($"{path}{startChar}\"{node.Segment}\"{endChar} --> {SanitizeMermaidNode(child.Value.Path)}{childStartChar}\"{child.Key}\"{childEndChar}");
+                ProcessNode(child.Value, writer);
+            }
+            if (String.IsNullOrEmpty(methods)) methods = "OTHER";
+            writer.WriteLine($"class {path} {methods}");
+        }
+
+        private static string GetMethods(OpenApiUrlTreeNode node)
+        {
+            return String.Join("_", node.PathItems.SelectMany(p => p.Value.Operations.Select(o => o.Key))
+                .Distinct()
+                .Select(o => o.ToString().ToUpper())
+                .OrderBy(o => o)
+                .ToList());
+        }
+
+        private static (string, string) GetShapeDelimiters(string methods)
+        {
+            
+            if (MermaidNodeStyles.ContainsKey(methods))
+            {
+                //switch on shape
+                switch (MermaidNodeStyles[methods].Shape)
+                {
+                    case MermaidNodeShape.Circle:
+                        return ("((", "))");
+                    case MermaidNodeShape.RoundedCornerRectangle:
+                        return ("(", ")");
+                    case MermaidNodeShape.Rhombus:
+                        return ("{", "}");
+                    case MermaidNodeShape.SquareCornerRectangle:
+                        return ("[", "]");
+                    case MermaidNodeShape.OddShape:
+                        return (">", "]");
+                    default:
+                        return ("[", "]");
+                }
+            }
+            else
+            {
+                return ("[", "]");
+            }
+        }
+        private static string SanitizeMermaidNode(string token)
+        {
+            return token.Replace("\\", "/")
+                    .Replace("{", ":")
+                    .Replace("}", "")
+                    .Replace(".", "_")
+                    .Replace(";", "_")
+                    .Replace("-", "_")
+                    .Replace("default", "def_ault");  // default is a reserved word for classes
+        }
+    }
+    /// <summary>
+    /// Defines the color and shape of a node in a Mermaid graph diagram
+    /// </summary>
+    public class MermaidNodeStyle
```

Comment:
does it need to be public?

Reply:
Yes if we are to make it customizable.

Reply:
then can you add it to the public APIs please?

Reply:
I changed my mind.  After playing around with the diagrams for a bit, I think the value of having consistency in diagram styles is more than the ability to customize.  We can always change our minds in the future.